### PR TITLE
add contentType and cacheControl headers

### PR DESCRIPTION
**Pull request by paulius-valiunas**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -17,4 +17,15 @@
     "**/lib": true,
     "common/temp": true
   },
+  "eslint.workingDirectories": [
+    {
+      "mode": "auto"
+    }
+  ],
+  // "eslint.format.enable": true,
```
Comment (austeja-bentley) [inline .vscode/settings.json:25]: why is this commended out?

Comment (paulius-valiunas) [reply .vscode/settings.json:25]: bad copypasta. It's not necessary, I removed it.

File: tests/backend-storage/src/test/utils/Helpers.ts [hunk 2]
```diff
@@ -48,6 +48,22 @@ export async function queryAndAssertContentEncoding(
   expect(contentEncoding).to.equal(expectedHeaders.contentEncoding);
 }
 
+export async function queryAndAssertCacheControl(
+  reference: ObjectReference,
+  expectedHeaders: ContentHeaders
+): Promise<void> {
+  const { cacheControl } = await serverStorage.getObjectProperties(reference);
+  expect(cacheControl).to.equal(expectedHeaders.contentEncoding);
```
[the remark this answers is not in the document]
Comment (paulius-valiunas) [reply tests/backend-storage/src/test/utils/Helpers.ts:56]: fixed

File: storage/s3/src/server/wrappers/S3ClientWrapper.ts [hunk 3]
```diff
@@ -102,6 +104,7 @@ export class S3ClientWrapper {
         Body: data,
         Metadata: metadata,
         ContentEncoding: headers?.contentEncoding,
+        CacheControl: headers?.cacheControl,
```
[the remark this answers is not in the document]
Comment (paulius-valiunas) [reply storage/s3/src/server/wrappers/S3ClientWrapper.ts:107]: fixed

File: utils/common-config/package.json [hunk 4]
```diff
@@ -26,6 +26,7 @@
   },
   "devDependencies": {
     "@itwin/eslint-plugin": "~3.5.1",
+    "@rushstack/eslint-patch": "^1.2.0",
```
[the remark this answers is not in the document]
Comment (paulius-valiunas) [reply utils/common-config/package.json:29]: the vscode eslint extension applies a patch to eslint that messes with module resolution, especially in rush monorepos. Without this patch it wouldn't be able to resolve any eslint plugins

**Pull request merged**